packages/api/src/types.ts

```diff
@@ -59,6 +59,7 @@ export interface QueryableStorageFunctionBase<CodecResult, SubscriptionResult> e
   hash: (arg?: CodecArg) => HashResult<CodecResult, SubscriptionResult>;
   key: (arg?: CodecArg) => string;
   size: (arg?: CodecArg) => U64Result<CodecResult, SubscriptionResult>;
+  head: () => CodecResult;
```

Comment:
I am unsure what should be interface for the `enumerate` method. Does it make sense to be `Iterator<CodecResult>`? Also is it helpful to expose lower level methods? Such as get head key, get next key, get previous key.

Reply:
Enumerate on client side needs to be very careful because the block head could change while enumerating. In that case I think I need to include the current block head in the Iterator?

Reply:
And `Iterator` would make sense. However forcing everybody to iteration themselves is probably not such a good idea. So it would probably make more sense to have the expose function return all the entries. (With an, maybe optional, length limit - so iteration is not done by everybody)

Reply:
So externally `head`, `next`, etc. is probably not the best since it add overhead. Just make an `.all` or `.list` where we retrieve all by iterating. (Subscribe to head, subsequent entries can use `getStorage`)